coreneuron/nrniv/nrn_setup.cpp

```diff
@@ -1137,10 +1142,26 @@ void read_phase2(FileHandler& F, int imult, NrnThread& nt) {
         nmech = F.read_int();
         tml_index = new int[nmech];
         ml_nodecount = new int[nmech];
+        int diff_modfiles_count = 0;
         for (int i = 0; i < nmech; ++i) {
             tml_index[i] = F.read_int();
             ml_nodecount[i] = F.read_int();
+            if (nrnmpi_myid == 0 &&
+                std::find(different_mechanism_type.begin(), different_mechanism_type.end(),
+                          tml_index[i]) != different_mechanism_type.end()) {
+                printf("Error: %s is a different MOD file than used by NEURON!\n",
+                       nrn_get_mechname(tml_index[i]));
+                diff_modfiles_count++;
+            }
```

Comment:
you were right, this if block doesn't look good to read...

you can compute error count from all rank but just do printf error conditional

Reply:
Okay. I will change all the if statements so that all ranks make the check but only rank 0 prints the messages. :+1: